core/src/main/java/com/graphhopper/routing/weighting/custom/ExpressionBuilder.java

```diff
@@ -0,0 +1,382 @@
+                parseBlockStatements();
+    }
+
+    interface CodeBuilder {
+        String create(Number n);
+    }
+
+    /**
+     * Injects the already parsed expressions (converted to BlockStatement) via janinos DeepCopier to the provided
+     * CompilationUnit cu (a class file).
+     */
+    private static Java.CompilationUnit injectStatements(List<Java.BlockStatement> priorityStatements,
+                                                         List<Java.BlockStatement> speedStatements,
+                                                         Java.CompilationUnit cu) throws CompileException {
+        cu = new DeepCopier() {
+            @Override
+            public Java.FieldDeclaration copyFieldDeclaration(Java.FieldDeclaration subject) throws CompileException {
+                // for https://github.com/janino-compiler/janino/issues/135
+                Java.FieldDeclaration fd = super.copyFieldDeclaration(subject);
+                fd.setEnclosingScope(subject.getEnclosingScope());
+                return fd;
+            }
+
+            @Override
+            public Java.MethodDeclarator copyMethodDeclarator(Java.MethodDeclarator subject) throws CompileException {
+                if (subject.name.equals("getSpeed") && !speedStatements.isEmpty()) {
+                    return injectStatements(subject, this, speedStatements);
+                } else if (subject.name.equals("getPriority")) {
+                    return injectStatements(subject, this, priorityStatements);
+                } else {
+                    return super.copyMethodDeclarator(subject);
+                }
+            }
+        }.copyCompilationUnit(cu);
+        return cu;
+    }
+
+    private static Java.MethodDeclarator injectStatements(Java.MethodDeclarator subject, DeepCopier deepCopier,
+                                                          List<Java.BlockStatement> statements) {
+        try {
+            if (statements.isEmpty())
+                throw new IllegalArgumentException("Statements cannot be empty when copying method");
+            Java.MethodDeclarator methodDecl = new Java.MethodDeclarator(
+                    new Location("m1", 1, 1),
+                    subject.getDocComment(),
+                    deepCopier.copyModifiers(subject.getModifiers()),
+                    deepCopier.copyOptionalTypeParameters(subject.typeParameters),
+                    deepCopier.copyType(subject.type),
+                    subject.name,
+                    deepCopier.copyFormalParameters(subject.formalParameters),
+                    deepCopier.copyTypes(subject.thrownExceptions),
+                    deepCopier.copyOptionalElementValue(subject.defaultValue),
+                    deepCopier.copyOptionalStatements(statements)
+            );
+            statements.forEach(st -> st.setEnclosingScope(methodDecl));
+            statements.clear();
```

Comment:
I wouldn't expect the `injectStatements` method to clear the collection here.

Reply:
It just makes sure the statements are not injected twice. But it is surely confusing. Will try something else.